contracts/MultitokenCollector.sol

```diff
@@ -0,0 +1,136 @@
+// SPDX-License-Identifier:MIT
+pragma solidity ^0.6.12;
+pragma experimental ABIEncoderV2;
+
+import "./interfaces/ICollector.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/math/SafeMath.sol";
+
+contract MultitokenCollector is ICollector {
+    address private _remainderAddress;
+    RevenuePartner[] private _partners;
+    IERC20[] private _tokens;
+    address public owner;
+
+    modifier onlyOwner() {
+        require(msg.sender == owner, "Only owner can call this");
+        _;
+    }
+
+    modifier noBalanceToShare() {
+        for (uint256 i = 0; i < _tokens.length; i++) {
+            require(
+                _tokens[i].balanceOf(address(this)) < _partners.length,
+                "There is balance to share"
+            );
+        }
+        _;
+    }
+
+    modifier updateValidShares(RevenuePartner[] memory partners) {
+        _;
+        uint256 totalShares;
+        for (uint256 i = 0; i < partners.length; i++) {
+            require(partners[i].share > 0, "0 is not a valid share");
+            totalShares = totalShares + partners[i].share;
+            _partners.push(partners[i]);
+        }
+        require(totalShares == 100, "Shares must add up to 100%");
+    }
+
+    constructor(
+        address _owner,
+        IERC20[] memory tokens,
+        RevenuePartner[] memory partners,
+        address remainderAddress
+    ) public updateValidShares(partners) {
+        owner = _owner;
+        _remainderAddress = remainderAddress;
+
+        for (uint i = 0; i < tokens.length; i++) {
+            _tokens.push(tokens[i]);
+        }
+    }
+
+    function getPartners() external view returns (RevenuePartner[] memory) {
+        return _partners;
+    }
+
+    function updateShares(
+        RevenuePartner[] memory partners
+    ) external onlyOwner noBalanceToShare updateValidShares(partners) {
+        delete _partners;
+    }
+
+    //@notice Withdraw the actual remainder and then update the remainder's address
+    //for a new one. This function is the only way to withdraw the remainder.
+    function updateRemainderAddress(
+        address remainderAddress
+    ) external onlyOwner noBalanceToShare {
+        for (uint256 i = 0; i < _tokens.length; i++) {
+            IERC20 token = _tokens[i];
+            uint256 balance = token.balanceOf(address(this));
+
+            if (balance != 0) {
+                // solhint-disable-next-line avoid-low-level-calls
+                (bool success, bytes memory ret) = address(token).call{
+                    gas: 200000
```

Comment:
is it necessary to have this hardcoded value here?

Reply:
Im not sure, I did some testing following the same link and its not necessary unless that we want to avoid sending extra gas, but over here is not possible. https://solidity-by-example.org/call/.  Therefore I think that we can remove it.

Reply:
ok will do then

Reply:
done! [ref](https://github.com/rsksmart/rif-relay-contracts/compare/72ebd56c074df9948ca62ba5f2b85b1e2b875fd2..b3a71a903eee79722f83118614f2af6cda6ff946)